src/player.cpp

```diff
@@ -6147,7 +6189,9 @@ void player::process_effects() {
 
             // Handle fatigue
             val = it.get_mod("FATIGUE", reduced);
-            if (val != 0) {
+            // Prevent ongoing fatigue effects while asleep.
+            // These are meant to change how fast you get tired, not how long you sleep.
+            if (val != 0 && !in_sleep_state()) {
```

Comment:
@Coolthulhu Sounds WAI to me.  Meth's fatigue effects will continue to happen while awake, this only affects what happens while asleep.  So, yes, if you use meth, you can go to bed and sleep through most of the ill effects.  Isn't that how it *should* work?

Wikipedia suggests that both sleeplessness and sleepiness are possible withdrawal effects, so if we really want to make the meth cooldown suck, we could make it inhibit sleep in addition to increasing tiredness.  So you'd have a nasty period where you're super-tired, but not actually able to fall asleep.

I'm fine with the coughing stuff.